target/linux/ath79/dts/qca9563_tplink_tl-wa1201-v2.dts

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca956x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,tl-wa1201-v2", "qca,qca9563";
+	model = "TP-Link TL-WA1201 v2";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		mdio-gpio0 = &mdio2;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "green:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wlan2g {
+			label = "green:wlan2g";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wlan5g {
+			label = "green:wlan5g";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		lan {
+			label = "green:lan";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "WPS button";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	mdio2: mdio {
+		compatible = "virtual,mdio-gpio";
+
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gpios = <&gpio 3 GPIO_ACTIVE_HIGH>, /* MDC */
+			<&gpio 4 GPIO_ACTIVE_HIGH>; /* MDIO */
+
+		phy4: ethernet-phy@4 {
+			reg = <4>;
+			device_type = "ethernet-phy";
```

Comment:
Do we need device_type, it's only used a single time in entire ath79 ...?

Reply:
So I went through the linux `dts` documentation they have schema of what should go into which dts node.
Following this sequence:
1. https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/mdio-gpio.yaml#L15
includes
2. https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/mdio.yaml
points to
3. https://github.com/torvalds/linux/blob/master/Documentation/devicetree/bindings/net/mdio.yaml#L62 which doesn't have this `device_type` node so It is probably superfluous.

On the other hand the last schema doesn't mention about nodes like `eee-broken-100tx` which are all over the openwrt codebase at this exact location and they work for sure.

This leads me to believe that schemata are not complete.

I'll remove `device_type`.